api/envoy/service/ext_proc/v3/external_processor.proto

```diff
@@ -170,6 +171,21 @@ message ProcessingResponse {
   // may use this to intelligently control how requests are processed
   // based on the headers and other metadata that they see.
   envoy.extensions.filters.http.ext_proc.v3.ProcessingMode mode_override = 9;
+
+  // When ext_proc server receives a request message, in case it needs more
+  // time to process the message, it sends back a ProcessingResponse message
+  // with a new timeout value. When Envoy receives this response message,
+  // it ignores other fields in the response, just stop the original timer,
+  // which has the timeout value specified in
+  // :ref:`message_timeout
+  // <envoy_v3_api_field_extensions.filters.http.ext_proc.v3.ExternalProcessor.message_timeout>`
+  // and start a new timer with this ``new_message_timeout`` value and keep the
+  // Envoy ext_proc filter state machine intact.
+  // Has to be >= 1ms and <=
+  // :ref:`max_message_timeout <envoy_v3_api_field_extensions.filters.http.ext_proc.v3.ExternalProcessor.max_message_timeout>`
+  // Such message can be sent at most once in a particular Envoy ext_proc filter processing state.
+  // To enable this API, one has to set ``max_message_timeout`` to a number >1ms.
+  google.protobuf.Duration new_message_timeout = 10;
```

Comment:
Do we want the complexity of these variable timeouts or would keepalive semantics work here?

Reply:
Thanks for the comments!
This new timeout is for special cases like WASM, which may need to cold start the software module after receiving the initial request from Envoy. And that cold start might take much longer (several seconds) than normal request processing (dozens milliseconds).  Just setting the normal timer :ref:`message_timeout <envoy_v3_api_field_extensions.filters.http.ext_proc.v3.ExternalProcessor.message_timeout>` to a number to cover the cold start case is considered to be not good, since it will cause Envoy waiting for too long in normal ext_proc request processing.

Reply:
OK. I think then rename this to `override_message_timeout` and LGTM.

Reply:
Good idea, Done!